### PR TITLE
Action packer re-use

### PR DESCRIPTION
**Pull request by jafri**
Been playing around with the tpl, let me know what you think @learnforpractice 

This might not be a good idea but just wanted to see what was possible

Modifies the action transformer to allow re-use (eliminating the @packer)

Before:
```
@packer
class MyData {
    constructor(
        public name: string
    ){}
}

const actionWrapper = new ActionWrapper("saygoodbye")
const action = actionWrapper.act(this.receiver, new PermissionLevel(this.receiver))
action.send(new MyData('alice'))
```

After
```
const actionWrapper = new ActionWrapper("saygoodbye")
const action = actionWrapper.act(this.receiver, new PermissionLevel(this.receiver))
action.send(new sayGoodbyeAction('alice'))
```

While this brings new simplicity to writing contracts, a downside is that you lose Intellisense on `new MyData` and instead have to rely on compile time checking

### CODE REVIEW

**Comment by learnforpractice** [comment]
Sounds good. I don't see a side effect.

**Review by learnforpractice — COMMENTED** [review]
Please add a test case if you can.

File: ts-packages/transform/src/preprocess/handlebars.ts [hunk 1]
```diff
@@ -34,19 +34,44 @@ Handlebars.registerHelper("generateActionMember", function (fn: ParameterNodeDef
...
+Handlebars.registerHelper("generateActionParam", function (fn: ParameterNodeDef) {
+    let code: string[] = [];
+    let plainType = fn.type.plainTypeNode;
+
+    if (plainType == 'string') {
+        code.push(` ${fn.name}: string = "",`);
+    } else if (plainType == 'string[]') {
+        code.push(` ${fn.name}: string[] = [],`);
+    } else {
+        if (TypeHelper.isPrimitiveType(fn.type.typeKind)) {
```
Comment (learnforpractice) [inline ts-packages/transform/src/preprocess/handlebars.ts:59]: primitive type may be a string, assigning 0 to a string will cause an error.

Comment (jafri) [reply ts-packages/transform/src/preprocess/handlebars.ts:59]: @learnforpractice doesnt the plainType check above for == 'string' cover that?

Comment (learnforpractice) [reply ts-packages/transform/src/preprocess/handlebars.ts:59]: Yes, ignore my previous comment. But I still smell that we need a refactor of this function. I will approve this PR, so we can iterate fast.

**Review by learnforpractice — APPROVED** [review]

**Comment by learnforpractice** [comment]
Solve the merge conflict please.

**Pull request merged**